### ISSUE TITLE
Case statements: Autocomplete does not complete enum names

### ISSUE THREAD

**Issue by jvilk-stripe** [sorbet/sorbet#5210]
#### Input

[→ View on sorbet.run](https://sorbet.run/?arg=--print=desugar-tree#%23%20typed%3A%20true%0A%0Aclass%20Blah%3A%3AFoo%20%3C%20T%3A%3AEnum%0A%20%20enums%20do%0A%20%20%20%20First%20%3D%20new%0A%20%20%20%20Second%20%3D%20new%0A%20%20end%0Aend%0A%0Aa%20%3D%20T.let%28Blah%3A%3AFoo%3A%3AFirst%2C%20Blah%3A%3AFoo%29%0A%0Acase%20a%0Awhen%20Blah%3A%3AFoo%3A%3A%0A%20%20%23%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%20Try%20to%20autocomplete%20here.%20No%20enum%20suggestions%20occur%0Aend%0A)

```ruby
# typed: true

class Blah::Foo < T::Enum
  enums do
    First = new
    Second = new
  end
end

a = T.let(Blah::Foo::First, Blah::Foo)

case a
when Blah::Foo::
  #             ^ Try to autocomplete here. No enum suggestions occur
end
```

#### Observed output

```
class <emptyTree><<C <root>>> < (::<todo sym>)
  class <emptyTree>::<C Blah>::<C Foo><<C <todo sym>>> < (<emptyTree>::<C T>::<C Enum>)
    <self>.enums() do ||
      begin
        <emptyTree>::<C First> = <self>.new()
        <emptyTree>::<C Second> = <self>.new()
      end
    end
  end

  a = <emptyTree>::<C T>.let(<emptyTree>::<C Blah>::<C Foo>::<C First>, <emptyTree>::<C Blah>::<C Foo>)

  begin
    <assignTemp>$2 = a
    if <emptyTree>::<C Blah>::<C Foo>::<C <ConstantNameMissing>>.===(<assignTemp>$2)
      <emptyTree>
    else
      <emptyTree>
    end
  end
end
editor.rb:13: expected constant name following "::" https://srb.help/2001
    13 |when Blah::Foo::
                      ^^
Errors: 1
```

Autocomplete menu lacks any enum values.

#### Expected behavior

Autocomplete menu displays enum values

- - -

This is a super common scenario since Sorbet supports exhaustive case statements over enum values.

**Comment by jez** [comment]
This looks like a weird interaction between the `===` desugaring and constant completion for `<ConstantNameMissing>`.

**Comment by jez** [comment]
@froydnj fixed this in #5015
